pytests/conftest.py

```diff
@@ -94,37 +118,40 @@ def wget(self, url, out, enforce_https=True, fingerprint=None):
         r.raw.decode_content = True
         with open(out, 'wb') as f:
             shutil.copyfileobj(r.raw, f)
-
         return True, None
 
     def run(self, cmd):
-        if cmd[0] is 'tdnf' and self.config['build_dir']:
-            cmd[0] = os.path.join(self.config['build_dir'], 'bin/tdnf')
+        if cmd[0] is 'tdnf':
+            if 'build_dir' in self.config:
+                cmd[0] = os.path.join(self.config['build_dir'], 'bin/tdnf')
+            if cmd[1] is not '--config':
```

Comment:
why dont we add -c for all commands without this check? do we have cases where we do not use the test config?

Reply:
When we are testing `--config` from tests, we must pass newly created config file (or dummy ones) to validate that the passed config is being considered. In those cases, this method must not pass any config file.